src/dev/flang/ast/AbstractFeature.java

```diff
@@ -1730,7 +1730,7 @@ public Expr visit(FeatureVisitor v, AbstractFeature outer)
    */
   public String toString()
   {
-    return visibility() + " " +
+    return (state().atLeast(State.FINDING_DECLARATIONS) ? visibility() + " " : "") +
```

Comment:
Would be much better to move this special handling to `visibility()` which already has a `NYI:` comment when `outer()` is called.

Reply:
@fridis I agree :-)